mutiny-core/src/vss.rs

```diff
@@ -0,0 +1,273 @@
+        let decrypted: Vec<u8> = cipher.decrypt_padded_vec_mut::<Pkcs7>(bytes).unwrap();
+        let decrypted_value = String::from_utf8(decrypted).unwrap();
+        let value = serde_json::from_str(&decrypted_value).unwrap();
+
+        VssKeyValueItem {
+            key: self.key,
+            value: Some(value),
+            version: self.version,
+        }
+    }
+}
+
+impl MutinyVssClient {
+    pub fn new(
+        auth_client: Arc<MutinyAuthClient>,
+        url: String,
+        encryption_key: SecretKey,
+        logger: Arc<MutinyLogger>,
+    ) -> Self {
+        Self {
+            auth_client,
+            url,
+            encryption_key,
+            logger,
+        }
+    }
+
+    pub async fn put_objects(&self, items: Vec<VssKeyValueItem>) -> Result<(), MutinyError> {
+        let url = Url::parse(&format!("{}/putObjects", self.url)).map_err(|e| {
+            log_error!(self.logger, "Error parsing put objects url: {e}");
+            MutinyError::Other(anyhow!("Error parsing put objects url: {e}"))
+        })?;
+
+        // check value is defined for all items
+        for item in &items {
+            if item.value.is_none() {
+                return Err(MutinyError::Other(anyhow!(
+                    "Value must be defined for all items"
+                )));
+            }
+        }
+
+        let items = items
+            .into_iter()
+            .map(|item| item.encrypt(&self.encryption_key))
+            .collect::<Vec<_>>();
+
+        // todo do we need global version here?
+        let body = json!({ "transaction_items": items });
+
+        self.auth_client
+            .request(Method::PUT, url, Some(body))
+            .await?;
+
+        Ok(())
+    }
+
+    pub async fn get_object(&self, key: &str) -> Result<VssKeyValueItem, MutinyError> {
+        let url = Url::parse(&format!("{}/getObject", self.url)).map_err(|e| {
+            log_error!(self.logger, "Error parsing get objects url: {e}");
+            MutinyError::Other(anyhow!("Error parsing get objects url: {e}"))
+        })?;
+
+        let body = json!({ "key": key });
+        let result: EncryptedVssKeyValueItem = self
+            .auth_client
+            .request(Method::POST, url, Some(body))
+            .await?
+            .json()
+            .await
+            .map_err(|e| {
+                log_error!(self.logger, "Error parsing get objects response: {e}");
+                MutinyError::Other(anyhow!("Error parsing get objects response: {e}"))
+            })?;
+
+        Ok(result.decrypt(&self.encryption_key))
+    }
+
+    pub async fn list_key_versions(
+        &self,
+        key_prefix: Option<String>,
+    ) -> Result<Vec<VssKeyValueItem>, MutinyError> {
+        let url = Url::parse(&format!("{}/listKeyVersions", self.url)).map_err(|e| {
+            log_error!(self.logger, "Error parsing list key versions url: {e}");
+            MutinyError::Other(anyhow!("Error parsing list key versions url: {e}"))
+        })?;
```

Comment:
If this is just for keys and version numbers, should we create just a type for this then? That would solve the optional value problem too. Seems like it'll only be an option based on the results here.